packages/irc2as/src/index.test.js

```diff
@@ -0,0 +1,87 @@
+const chai = require('chai');
+const sinon = require('sinon');
+const schemas = require('sockethub-schemas');
+const fs = require("fs");
+const equal = require('fast-deep-equal');
+const expect = chai.expect;
+
+const IRC2AS = require('./index');
+const outputs = require('./index.test.data');
+const ircdata = fs.readFileSync('./src/index.test.data.irc.txt', 'utf-8');
+const inputs = ircdata.split('\n');
+
+function matchStream(done) {
+  return (stream) => {
+    expect(typeof stream.published).to.eql('string');
+    delete stream.published;
+    let matched = false;
+    for (let i = 0; i <= outputs.length; i++) {
+      matched = equal(stream, outputs[i]);
+      if (matched) {
+        outputs.splice(i, 1);
+        break;
+      }
+    }
+    if (! matched) {
+      console.log();
+      console.log('available matches:' + JSON.stringify(outputs));
+      console.log('failed to find match for: ' + JSON.stringify(stream));
+      return done(new Error('failed matching ' + JSON.stringify(stream)));
+    }
+    const err = schemas.validator.validateActivityStream(stream);
+    if (err) {
+      done(err + ' - ' + JSON.stringify(stream));
+    }
+  }
+}
+
+describe("irc2as converts incoming IRC data to ActivityStreams", () => {
```

Comment:
This is the text that belongs in an "it" expectation. Conversely, the "it" text below is a "describe" text actually. If you actually use context and expectation blocks as intended, you will see that it also helps with better structuring specs, since "it" should mostly just contain the expectations (converts x to y), and multiple of them oftentimes, while "describe" should only describe *what* is being tested, not what the expectations are.

See here for a better explanation with example: https://stackoverflow.com/a/32055428